### ISSUE TITLE
SPFx 1.17.1 - custom dialog

### ISSUE THREAD

**Issue by necik11** [SharePoint/sp-dev-docs#8895]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
  - Chrome 112.0.5615.86
- SPFx version
  - 1.17.1  
- Node.js version
  - v16.14.1

### Describe the bug / error

I would like to report/reopen a bug with the dialog box even after upgrading the SPFx version...
[Use custom dialog boxes with SharePoint Framework Extensions](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/extensions/guidance/using-custom-dialogs-with-spfx)

in order to confirm this, I created a new project and went step by step:
1. Make new project **spfx-pdfjs**
2. Create custom dialog import into my Command and woala: 

- Could not load hello-world-command-set in require. TypeError: Cannot read properties of undefined (reading 'id')
- ![image](https://user-images.githubusercontent.com/25322303/231988080-e86073cb-a814-448e-9b94-bfd68b8ef3a7.png)
- [Closed issue #8795](https://github.com/SharePoint/sp-dev-docs/issues/8795)
- npm ls react:
spfx-pdfjs@0.0.1 C:\github\spfx-pdfjs\spfx-pdfjs
├─┬ @microsoft/sp-core-library@1.17.1
│ ├─┬ react-dom@17.0.2
│ │ └── react@17.0.2 deduped
│ └── react@17.0.2
└─┬ @microsoft/sp-dialog@1.17.1
  ├─┬ @fluentui/react@8.107.5
  │ ├─┬ @fluentui/foundation-legacy@8.2.33
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/react-focus@8.8.19
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/react-hooks@8.6.20
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/react-portal-compat-context@9.0.5
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/react-window-provider@2.2.9
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/theme@2.6.25
  │ │ └── react@17.0.2 deduped
  │ ├─┬ @fluentui/utilities@8.13.9
  │ │ └── react@17.0.2 deduped
  │ └── react@17.0.2 deduped
  ├─┬ @microsoft/sp-application-base@1.17.1
  │ └─┬ @microsoft/sp-loader@1.17.1
  │   ├─┬ react-dom@17.0.1
  │   │ └── react@17.0.1 deduped
  │   └── react@17.0.1
  ├─┬ react-dom@17.0.1
  │ └── react@17.0.1 deduped
  └── react@17.0.1
- so i still dont understand why i should use 
-` npm i react@17.0.1 react-dom@17.0.1 @types/react@17.0.45 @types/react-dom@17.0.17 --save-exact`
- although it's a clean install/generate new project and instructions straight from the documentation?

I make [repository](https://github.com/necik11/spfx-pdfjs), but i added just CustomDialog from tutorial:

 
 

### Steps to reproduce

Provided in details

### Expected behavior

working without reinstall packages versions

**Comment by s-KaiNet** [comment]
Actually, we have exactly the same problem by simply upgrading from 1.17.0 to 1.17.1. Application customer doesn't load with

> 
>  ERROR:
> Cannot read properties of undefined (reading 'id')
> 
> CALL STACK:
> TypeError: Cannot read properties of undefined (reading 'id')
>     at https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_380255c8e20811346b873b876a9bc242.js:70:747393
>     at new Promise (<anonymous>)
>     at e.loadComponent (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_380255c8e20811346b873b876a9bc242.js:70:747361)
>     at https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_380255c8e20811346b873b876a9bc242.js:70:749122
>     at async Promise.all (index 0)

The console output:
![image](https://user-images.githubusercontent.com/10658276/233599458-c689957d-57fc-4b1e-a01b-b7f9d260dec0.png)

When migrating back to 1.17.0 everything starts working.

**Comment by necik11** [comment]
I don't know for sure, but I have a feeling that I tried it on version 1.17.0 and it did the same error in this version. So it's probably lucky if it worked for you here initially...
Anyway (**workaround**) command, it should still work:
`npm i react@17.0.1 react-dom@17.0.1 @types/react@17.0.45 @types/react-dom@17.0.17 --save-exact`
but for the future it would be good to fix it.

**Comment by s-KaiNet** [comment]
I have these dependencies already in my package.json, thus this workaround doesn't work for me. I believe SPFx 1.17 automatically adds those dependencies.

**Comment by necik11** [comment]
do you try `gulp serve` locally or directly `gulp package --ship` ?

Because right now I'm trying something on this version and this workaround works... - for now I'm trying gulp serve locally.

maybe you need to delete node_modules +package-lock.json -> npm install at end call `gulp build`  ->` gulp serve`

**Comment by s-KaiNet** [comment]
The thing is, that everything works locally, but not when you deploy it.

**Comment by necik11** [comment]
Everything works fine for me... maybe really need delete node_modules and package-lock.json... and again.
But yes I didn't make an upgrade- i create new one project... cause it was much easier...

**Comment by AJIXuMuK** [comment]
For extensions (app customizer and command set) React is not added to dependencies. And you need to go and add (`npm i`) them.
We should update the docs soon to point it out as, unfortunately, it wasn't in the docs before.

@s-KaiNet - in your case it can't load one of dependencies (weird - it's masked here on GH, but the screenshot in my email still shows the name).
I would double-check if for some reason config for this dependency has changed.

**Comment by s-KaiNet** [comment]
Looks like mystery solved.  
With SPFx upgrade, we also upgraded the version of [ `mgt-spfx`](https://github.com/microsoftgraph/microsoft-graph-toolkit) version to 2.10. But it seems [something went totally wrong](https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/2205) with the release of SPFx package for 2.10 version of mgt. When I reverted `mgt-spfx` to `2.9.0`, everything started to work with SPFx 1.17.1. From the same time it worked locally even with  `mgt-spfx@2.10.0` most likely because the dependencies were taken from `node_modules`, but upon publishing the app catalog `mgt-spfx` package is used (and has some issues). So it's not SPFx issue in our case.
